lib/gooddata/core/splunk_logger.rb

```diff
@@ -0,0 +1,193 @@
+        filter << "(" + Regexp.quote(val) + ")" if val.class == String
+        filter.concat filter_from_param_array(val) if val.class == Array
+        filter.concat filter_from_param_array(val.values) if val.class == Hash
+      end
+      filter
+    end
+
+    def mode_on(mode)
+      @modes[mode] = true
+    end
+
+    def mode_off(mode)
+      @modes[mode] = false
+    end
+
+    [:buffering, :api_output, :file_output].each do |mode|
+      define_method :"#{mode}_on" do
+        mode_on(mode)
+      end
+
+      define_method :"#{mode}_off" do
+        mode_off(mode)
+      end
+
+      define_method :"#{mode}_on?" do
+        @modes[mode]
+      end
+    end
+
+    def time_from_action_start(t)
+      (t - @action_start) * 1000
+    end
+
+    def start_action(action)
+      flush_aggregator
+
+      now = Time.now
+      end_action
+      set_context(ACTION_CONTEXT, action)
+      @action_start = now
+      set_context ACTION_STATUS_CONTEXT, 'start'
+      add INFO, nil, '', now
+      set_context ACTION_STATUS_CONTEXT, 'in_progress'
+    end
+
+    def end_action(now = Time.now)
+      return if @context[:action] == 'undefined' || @context[:status] == 'outside'
+      flush_aggregator
+
+      set_context ACTION_STATUS_CONTEXT, 'end'
+      add INFO, nil, '', now
+      set_context ACTION_STATUS_CONTEXT, 'outside'
+      set_context ACTION_CONTEXT, 'undefined'
+    end
+
+    def set_context(key, val)
+      @context[key] = val
+    end
+
+    def clear_logs
+      @logs = ""
+    end
+
+    def add(severity, message = nil, progname = nil, time = nil)
+      severity ||= UNKNOWN
+      if @logdev.nil? or severity < @level
+        return true
+      end
+      progname ||= @progname
+      if message.nil?
+        if block_given?
+          message = yield
+        else
+          message = progname
+          progname = @progname
+        end
+      end
+
+      now = (time.nil? ? Time.now : time)
+
+      context = SplunkLogger.hash_to_string(@context.merge(:time => time_from_action_start(now)))
+      if message.class == Hash
+        message_formatted = SplunkLogger.hash_to_string(message) + context
+      elsif message.nil?
+        message_formatted = context
+      else
+        message_formatted = message.to_s + context
+      end
+
+      severity_formatted = format_severity severity
+
+      if buffering_on?
+        @logs << @formatter.call(severity_formatted, now, progname, message_formatted).to_s
+      else
+        GoodData.splunk_api_helper.send_logs @formatter.call(severity_formatted, now, progname, message_formatted).to_s if api_output_on?
+        super(severity, message_formatted, progname) if file_output_on?
+      end
+    end
+
+    def flush
+      unless @logs.empty?
+        GoodData.splunk_api_helper.send_logs @logs if api_output_on?
+        @logdev.write @logs if file_output_on? && !@logdev.nil?
+        clear_logs
+      end
+    end
+
+    def aggregate(method, endpoint, duration, domain='undefined')
```

Comment:
Style/SpaceAroundEqualsInParameterDefault: Surrounding space missing in default value assignment.